raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -146,7 +149,7 @@ contract TokenNetwork is Utils {
 
     modifier isOpen(address participant, address partner) {
         uint256 channel_identifier = getChannelIdentifier(participant, partner);
-        require(channels[channel_identifier].state == 1);
+        require(channels[channel_identifier].state == uint8(ChannelState.Opened));
```

Comment:
From what I know, `ChannelState` should actually be `uint8`. Did you have a compilation issue?

> "The integer type used is just large enough to hold all enum values" 
https://solidity.readthedocs.io/en/v0.4.24/types.html#enums

Reply:
from the docs linked above:

> They are explicitly convertible to and from all integer types but implicit conversion is not allowed.

Reply:
We can remove the explicit casts from the assignments to the `Channel` struct if we change [`uint8 state` to `ChannelState state`](https://github.com/raiden-network/raiden-contracts/blob/53d3c10971a994e77dc5cf463bf89f4e61e118c1/raiden_contracts/contracts/TokenNetwork.sol#L81), but then we would have to add the casts on all the events that use the channel id.

Reply:
I would prefer `ChannelState state;` inside the `Channel` struct with casting in `getChannelInfo` 

>  we would have to add the casts on all the events that use the channel id.

Why?

Reply:
> And you don't need casting in getChannelInfo, because you can have returns (uint256, ChannelState)

This is not support with the current ABI, only with the experimental.

Reply:
We are talking about the channel state now.
Are you referring to https://github.com/raiden-network/raiden-contracts/issues/210?

`struct` as a function arguments is a recent addition. Until recently, only `ethers.js` supported this. I have not tested yet with `web3.py`

Reply:
> This is not supported with the current ABI, only with the experimental.

Tested only with `0.4.24+commit.e67f0147.Emscripten.clang`, without an additional `pragma` and it worked.

Reply:
> but then we would have to add the casts on all the events that use the channel id.

My bad, I errd. the channel state is **not** used in any of the events that we have now, so for the events this is not a problem.